src/drivers/bitbucket_cloud.js

```diff
@@ -22,39 +22,66 @@ class BitBucketCloud {
 
-    // Make a comment in the commit
     const commitEndpoint = `/repositories/${projectPath}/commit/${commitSha}/comments/`;
-    const commitBody = JSON.stringify({ content: { raw: report } });
-    const commitOutput = await this.request({
-      endpoint: commitEndpoint,
-      method: 'POST',
-      body: commitBody
-    });
+
+    const existingCommmit = (
+      await this.paginatedRequest({ endpoint: commitEndpoint, method: 'GET' })
+    )
+      .filter(
+        (comment) =>
+          comment.content.raw && comment.content.raw.endsWith(watermark)
+      )
+      .sort((first, second) => first.id < second.id)
+      .pop();
+
+    const commitOutput = (
+      await this.request({
+        endpoint:
+          commitEndpoint +
+          (update && existingCommmit ? existingCommmit.id : ''),
+        method: update && existingCommmit ? 'PUT' : 'POST',
+        body: JSON.stringify({ content: { raw: report } })
+      })
+    ).links.html.href;
 
     // Check for a corresponding PR. If it exists, also put the comment there.
-    const getPrEndpt = `/repositories/${projectPath}/commit/${commitSha}/pullrequests`;
-    const { values: prs } = await this.request({ endpoint: getPrEndpt });
+    let prs;
+    try {
+      const getPrEndpoint = `/repositories/${projectPath}/commit/${commitSha}/pullrequests`;
+      prs = await this.paginatedRequest({ endpoint: getPrEndpoint });
+    } catch (err) {
+      if (err.message === 'Not Found Resource not found')
+        console.warn(
+          "Can't create a pull request comment: the Pull Request Commit Links application has not been installed."
+        );
+      else throw err;
+    }
 
     if (prs && prs.length) {
       for (const pr of prs) {
-        try {
-          // Append a watermark to the report with a link to the commit
-          const commitLink = commitSha.substr(0, 7);
-          const longReport = `${commitLink}   \n${report}`;
-          const prBody = JSON.stringify({ content: { raw: longReport } });
-
-          // Write a comment on the PR
-          const prEndpoint = `/repositories/${projectPath}/pullrequests/${pr.id}/comments`;
-          await this.request({
-            endpoint: prEndpoint,
-            method: 'POST',
-            body: prBody
-          });
-        } catch (err) {
-          console.debug(err.message);
-        }
+        // Append a watermark to the report with a link to the commit
+        const commitLink = commitSha.substr(0, 7);
+        const longReport = `${commitLink}   \n${report}`;
+        const prBody = JSON.stringify({ content: { raw: longReport } });
+
+        // Write a comment on the PR
+        const prEndpoint = `/repositories/${projectPath}/pullrequests/${pr.id}/comments/`;
+        const existingPr = (
+          await this.paginatedRequest({ endpoint: prEndpoint, method: 'GET' })
+        )
+          .filter(
+            (comment) =>
+              comment.content.raw && comment.content.raw.endsWith(watermark)
```

Comment:
can we deestructure here?
```js
const { content: { raw = '' } = {} } = comment;
raw && raw.endsWith(watermark)
```

Reply:
Yes, we can, but I don't see any advantage over the current approach, which, to my mind, seems a lot clearer. Feel free to unresolve if you think otherwise.

Reply:
Nope. Its js style and we have to stick with a style. Aside that is far larger.

Reply:
As you mentioned [here](https://iterativeai.slack.com/archives/C01900GSB4J/p1623844680401200), we're using [this convention](https://github.com/airbnb/javascript#destructuring).  Nevertheless, I'm failing to see how the proposed solution could be shorter or more readable than the original.

```suggestion
              {
                const { content: { raw = '' } = {} } = comment;
                return raw.endsWith(watermark);
              }
```

Feel free to update the code under your own responsibility. ☠️  I still think that [this](https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds) prevails over the style guide. 😉

Reply:
> Nevertheless, I'm failing to see how the proposed solution could be shorter or more readable than the original.

shorter the line @0x2b3bfa0 and dryer

```js
myobject.oneproperty.amazing && myobject.oneproperty.amazing.istrulyamazing && myobject.oneproperty.amazing.istrulyamazing()
```

its way larger than having the solution destructured. This is something that tends to happen in Object based languages and JS is an Object based language (where even a function is an Object). To avoid this ES6 included deestructuring  as a first citizen. So even in a case where you don't see the benefit of it, being consistent allows you to write it without think on it. 

P.D. Im  very sorry to have a little mind

Reply:
> _and dryer_

Ouch! 😄  That's unquestionably true.

> _its way larger than having the solution destructured. This is something that tends to happen in Object based languages and JS is an Object based language (where even a function is an Object). To avoid this ES6 included deestructuring as a first citizen. So even in a case where you don't see the benefit of it, being consistent allows you to write it without think on it._

Once and if we move to Node JS 14+ we can use `comment?.content?.raw?.endsWith(watermark)` thanks to [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining); it's the really idiomatic solution to this issue.

Anyhow, I don't want to spend your time in sterile discussions. 🤷🏼  If you think it's the way to go, I'm fine with it.

> _P.D. Im very sorry to have a little mind_

Don't take it personal, it's about approaches, not about minds.

Reply:
> Anyhow, I don't want to spend your time in sterile discussions. 🤷🏼 If you think it's the way to go, I'm fine with it.

It's good to spend time on this. To be working on the same page... its never about me or my personal preferences. If we don't like the style lets chose one and stick with it.

Reply:
> _It's good to spend time on this. To be working on the same page... its never about me or my personal preferences. If we don't like the style lets chose one and stick with it._

Awesome! I don't want to impose my personal preferences either, as it would be as unfair and impractical; let's ping @casperdcl so we can take a more plural decision. Destructuring doesn't look that bad either, but it seemed a bit more Perl-ified than the original code. 

In this particular case, I would be really happy to use Node JS 14+ [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) instead of nested destructurings or infinite checking mantras. Nevertheless, we're still stuck on Node JS 12 and the immediate solution would require us to resort to any of the discussed approaches.

Reply:
we have #596 for this

Reply:
And https://github.com/iterative/cml/issues/597

Reply:
Closed with https://github.com/iterative/cml/issues/596#issuecomment-862491282 and https://github.com/iterative/cml/pull/581/commits/68cb5a27dbfccea85d9aa1dbe06287cea519c20c